pom.xml

```diff
@@ -527,7 +528,7 @@
                             <compilerArgs>
                                 <arg>-Xlint:all,-serial,-options,-path</arg>
                                 <arg>-parameters</arg>
-                                <arg>-Werror</arg>
+                                <arg>${compile.warnings-flag}</arg>
```

Comment:
Hey @colinhicks I think we'll need a similar change in some of the individual modules. Specifically, `ksqldb-engine`, `ksqldb-execution`, `ksqldb-parser`, `ksqldb-streams`, `ksqldb-benchmark`, and `ksqldb-udf-quickstart` specify overrides for this plugin which means `-Werror` is still hard-coded in those modules. 

I can inherit this PR from you if you're low on bandwidth. Just let me know :)

Reply:
Thanks for taking a look and good catch. I will update the modules, too

Reply:
Thanks for the updates! I think https://github.com/confluentinc/ksql/blob/master/ksqldb-udf-quickstart/src/main/resources/archetype-resources/pom.xml#L64 needs to be updated as well, though I haven't actually tried it.

Reply:
I don't think the archetype resources are necessary to update - that pom is used (if I understand correctly) to generate new poms for people building custom UDFs

Reply:
Yeah, I skipped archetype-resources because it seems to have a separate build lifecycle. I defer to you all on whether to remove `-Werror` there in this PR.

Reply:
🤷 let's punt on that for now

Reply:
@agavra what's the case for removing `-Werror` from the archetype-resources pom entirely?

Reply:
I feel like we shouldn't be forcing people who write UDFs to treat errors as warnings - that's not a standard compiler flag